### DISCUSSION TITLE
How do I plot ODFs using Euler angles?

### DISCUSSION THREAD

**Discussion by SingPr** [mtex-toolbox/mtex#1463] [Ask Anything] [answered]
Hi everyone:
   I am a beginner level Mtexer , here, I have a txt data of Euler angle, not an odf data. Now, I wanna plot an odf figure. What can I do?Any ideas how to do this in MTEX? Any reply would be appreciated!
Thanks！
Justin
My data:
phi1        phi2     phi3     weight
140.79   47.95    8.52   0.0001000
  140.63   48.10    8.78   0.0001000
  140.94   47.81    8.30   0.0001000
  140.71   48.02    8.66   0.0001000
  140.91   47.87    8.32   0.0001000
  140.91   47.84    8.34   0.0001000
  140.87   47.87    8.41   0.0001000
  141.02   47.74    8.15   0.0001000
  140.70   48.03    8.66   0.0001000
 -161.27   35.96   23.91   0.0001000
 -161.12   35.99   23.72   0.0001000
 -161.14   35.99   23.75   0.0001000
 -161.06   36.01   23.65   0.0001000
 -161.78   35.85   24.55   0.0001000
 -160.67   36.10   23.16   0.0001000
 -161.32   35.95   23.97   0.0001000
 -160.94   36.04   23.49   0.0001000
.........

**Comment by lfgmorales** [comment]
Hi Justin

First you have to identify the format of your data

You can start from here

https://mtex-toolbox.github.io/OrientationImport.html <https://mtex-toolbox.github.io/OrientationImport.html>

Particularly this line

ori=orientation.load(fname,’columnNames’,{‘phi1’,’phi2’,’phi3’,’weights'},cs)

Where fname is the name of your input file with these orientations and cs is the crystal symmetry


cheers

Luiz

**Reply by SingPr** [reply]
Hi, Luiz
  thanks a lot. I can plot a pole diagram, but odfs is difficult to plot, can you describe it in more detail or provide a demonstration? Thanks!
Best
Justin

**Comment by lfgmorales** [comment, marked as answer]
Hi

I would suggest you start from here


https://mtex-toolbox.github.io/ODFTheory.html <https://mtex-toolbox.github.io/ODFTheory.html>


Then maybe here to the pole figures you want

https://mtex-toolbox.github.io/EBSDOrientationPlots.html <https://mtex-toolbox.github.io/EBSDOrientationPlots.html>


cheers


Luiz

**Reply by SingPr** [reply]
Thank you, Luiz, it's help for me.
